xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/flamingo.js

```diff
@@ -40,14 +40,78 @@ require(['jquery', 'bootstrap'], function($) {
+  In order to create a drawer that is compatible with this script, two elements are necessary:
+  * A drawer toggler button, that will have:
+   ** a reference to the drawer container in its attribute 'aria-control'
+   ** the class 'drawer-toggle'
+  * A drawer container, that will have:
+   ** a unique ID
+   ** the class 'drawer-nav' (for style) 
+   ** an accessible name 
+   ** The content to display inside.
+  In addition, the drawer can contain a `.drawer-close` button that will close the drawer on click.
+  For an example of drawer creation, see #tmDrawerActivator and #tmDrawer.
+ */
+require(['jquery'], function($) {
+  $('.drawer-toggle').each(function(index) {
+    // Setting up the drawer.
+    let drawerContainerToggler = $(this);
+    let drawerId = drawerContainerToggler.attr('aria-controls');
+    let drawerContainer = $(document.getElementById(drawerId));
+    
+    let openDrawer = () => drawerContainer.trigger('drawer' + index + '.opened');
+    let closeDrawer = () => drawerContainer.trigger('drawer' + index + '.closed');
+    drawerContainerToggler.on('click', openDrawer);
+    // Close drawer when clicking the backdrop (or any element outside of the drawer itself).
+    drawerContainer.on('click', (event) => {
+      let drawerzone = event.target.getBoundingClientRect();
+      if (drawerzone.left > event.clientX || drawerzone.right < event.clientX
+        || drawerzone.top > event.clientY || drawerzone.bottom < event.clientY) {
+        closeDrawer();
+      }
+    });
+    // Close drawer when clicking on a close button inside it
+    drawerContainer.find('.drawer-close').on('click', closeDrawer);
+
+    drawerContainer.on('drawer' + index + '.opened', function(event) {
+      // We use the drawer-transitioning class to make sure the transition to slidein is not shortcutted when showing the modal
+      drawerContainer.addClass('drawer-transitioning');
+      drawerContainerToggler.attr('aria-expanded', 'true');
+      drawerContainer.get(0).showModal();
+      drawerContainer.removeClass('drawer-transitioning');
+      // The drawer can be closed by pressing the ESC key
+      $("body").on('keydown.drawerClose', function (event) {
+        if (event.key === 'Escape') {
+          closeDrawer();
+        }
+      });
+    }).on('drawer' + index + '.closed', function(event) {
+      drawerContainerToggler.attr('aria-expanded', 'false');
+      
+      function waitAnimation() {
+        drawerContainer.get(0).close();
+        drawerContainer.removeClass('drawer-transitioning');
+        drawerContainer.get(0).removeEventListener('transitionend', waitAnimation);
+      }
+      // We use the drawer-transitioning class to give time for the hide modal transition to play.
+      drawerContainer.addClass('drawer-transitioning');
+      drawerContainer.get(0).addEventListener('transitionend', waitAnimation);
+      // We remove the listener that was created when the drawer opened up
+      $("body").off('keydown.drawerClose');
```

Comment:
I wonder how that works with multiple drawers if one drawer is triggered from another drawer. Afaik we plan this use case for What's New.

Reply:
I added the index to the event name, so we're sure there's no collisions between drawers :)
Addressed in https://github.com/xwiki/xwiki-platform/pull/2282/commits/911f5323562fce20e818135e8cbd4c1500e04629